### PR TITLE
Issue #1582: RSS feed icon never printed

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/1582

### CODE REVIEW

**Comment by jenlampton** [comment]
Here's a crazy idea: what if we turned rss.xml into a feed display for the promoted view, and then replaced the hard-coded call to `theme_feed_icon()` with `backdrop_get_feeds()`? Would that be the same result?

File: core/modules/node/node.module [hunk 1]
```diff
@@ -2014,9 +2010,35 @@ function node_block_configure($delta = '', $settings = array()) {
       '#options' => backdrop_map_assoc(array(2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 25, 30)),
     );
   }
+  if ($delta == 'syndicate') {
+    $form['syndicate_settings'] = array(
+      '#type' => 'radios',
+      '#title' => t('Choose the linked feed'),
+      '#default_value' => isset($settings['syndicate_settings']) ? $settings['syndicate_settings'] : 'promoted',
+      '#options' => array(
+        'promoted' => t('Promoted nodes'),
```
Comment (jenlampton) [inline core/modules/node/node.module]: label: "Always show RSS link for promoted content"

File: core/modules/node/node.module [hunk 2]
```diff
@@ -2014,9 +2010,35 @@ function node_block_configure($delta = '', $settings = array()) {
...
+        'page_content' => t('Page content'),
```
Comment (jenlampton) [inline core/modules/node/node.module]: label: "Show RSS link for the current page, if available."

**Review by jenlampton — COMMENTED** [review]
I'm a little confused by these labels, how about something more like this?

**Comment by docwilmot** [comment]
Yes, I should have done some explaining. The original icon only showed a list of promoted nodes, which is not as useful now as when Drupal front page was just a list by default. So thought to make it more useful by allowing it to show the current page instead, especially useful if the current page is a View.

File: core/modules/node/node.module (hunk 1 above)
Comment (docwilmot) [reply core/modules/node/node.module]: fixed

File: core/modules/node/node.module (hunk 2 above)
Comment (docwilmot) [reply core/modules/node/node.module]: done

File: core/modules/node/node.module [hunk 3]
```diff
@@ -2014,9 +2010,35 @@ function node_block_configure($delta = '', $settings = array()) {
...
+ */
+function node_syndicate_content($settings) {
+  if (isset($settings['syndicate_settings']) && $settings['syndicate_settings'] == 'page_content') {
+    return backdrop_get_feeds();
+  }
+  // Default to promoted content to support legacy Syndicate blocks.
+  return array(
+    '#theme' => 'feed_icon',
+    '#url' => 'rss.xml',
+    '#title' => t('Syndicate'),
```
Comment (opi) [inline core/modules/node/node.module]: the `#title` is used to create the feed icon title attribute, then this one generate a tooltip with "Subscribe to Syndicate", maybe this PR is a good occasion to do better, something like "Subscribe to promoted content (feed)".

Comment (jenlampton) [reply core/modules/node/node.module]: Even just "Subscribe" would be better than "Syndicate". Would anyone miss the previous language here?

**Comment by quicksketch** [comment]
Merged into 1.x and 1.11.x. Thanks!

**Pull request merged**